### PR TITLE
Implemented x-forwarded-prefix proxy fix + test

### PR DESCRIPTION
**Pull request by ilansh**
This is my first contribution here.
Please look at issue #1237 
Although x-forwarded-prefix doesn't seem like an official standard, I think it would be nice to add support.

I added test, made sure it runs with the fix and fails without it.

### CODE REVIEW

**Comment by davidism** [comment]
Overall it looks good. You don't have git configured with your email registered with GitHub, so you might want to fix that.

**Comment by ilansh** [comment]
Thanks.
I think I fixed the email issue

**Comment by davidism** [comment]
Would you add a line in the test to check that it builds the expected url as well.

**Comment by ilansh** [comment]
@davidism  Added build and match tests.
I wasn't too sure how to test values individually, so the test is just string comparison (copied from another test), hope it's fine.

**Comment by davidism** [comment]
Are proxies guaranteed to strip all incoming `X-Forwarded` headers from the public? I'm hesitant to add this unguarded. Maybe an `__init__` arg `prefix=False` would be safer, and I could go and add args for the other headers too. That way the user can configure exactly what they're prefix supports.

File: werkzeug/contrib/fixers.py [hunk 1]
```diff
@@ -136,6 +136,7 @@ def __call__(self, environ, start_response):
         forwarded_proto = getter('HTTP_X_FORWARDED_PROTO', '')
         forwarded_for = getter('HTTP_X_FORWARDED_FOR', '').split(',')
         forwarded_host = getter('HTTP_X_FORWARDED_HOST', '')
+        forwarded_prefix = getter('HTTP_X_FORWARDED_PREFIX', '')
         environ.update({
             'werkzeug.proxy_fix.orig_wsgi_url_scheme':  getter('wsgi.url_scheme'),
             'werkzeug.proxy_fix.orig_remote_addr':      getter('REMOTE_ADDR'),
```
Comment (davidism) [inline werkzeug/contrib/fixers.py]: `werkzeug.proxy_fix.orig_script_name` should be set here.

**Comment by ilansh** [comment]
I guess proxies are not guaranteed to do anything with those headers.  
I could add a `prefix=False` arg to `__init__`, but regarding args for the other headers - they would have to default to `True` or we would break backwards compatibility - correct me if I'm wrong.

**Comment by ilansh** [comment]
I'm thinking maybe default `fwd_prefix` to True as well, to keep it consistent with the others.

**Comment by ilansh** [comment]
@davidism did you review the latest changes? let me know if anything else is needed here

**Pull request merged**